addon/-private/errors.js

```
@@ -0,0 +1 @@
+export class TypeError extends Error {}
```

Recently we had this pattern blow up in main, turns out extending errors is not generally a good idea. We should just throw an error with a good message instead, and maybe set an error code.

---

That's fine with me. I'll remove this and throw a regular error instead.